### PR TITLE
Publish User Crud with default fields

### PR DESCRIPTION
**Pull request by ShaileshInfyom**
### What Implemented? 

- User CRUD publish with default field which are provided by Laravel framework.
-  Publish User CRUD via using `php artisan infyom.publish:user` command

**following files generated by this command**

1. Controller
2. Repository
3. Views
4. Route
5. Menu
6. Requests

before fire this command need to complete installation steps and publish layouts

after publish user CRUD. 

for more detail about issue.
How to integrate Users with scaffold? [#666](https://github.com/InfyOmLabs/laravel-generator/issues/666)

#### How to Test?
Add this package in. composer and update it. do all step of infyom pkg installations. 
```
 "infyomlabs/laravel-generator": "dev-issue-666-user-publish",
  "infyomlabs/adminlte-templates": "dev-issue-666-user-crud-template",
```
then after run command   `php artisan infyom.publish:user`
Check admin panel. you can see user menu and test it.

### CODE REVIEW

File: src/Commands/Publish/PublishUserCommand.php [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+    private function createDirectories($dir)
+    {
+        FileUtil::createDirectoryIfNotExist($dir);
+    }
+
+    private function getViews()
+    {
+        return ['users/create' => 'users/create.blade.php', 'users/edit' => 'users/edit.blade.php', 'users/fields' => 'users/fields.blade.php', 'users/index' => 'users/index.blade.php', 'users/show' => 'users/show.blade.php', 'users/show_fields' => 'users/show_fields.blade.php', 'users/table' => 'users/table.blade.php'];
```
Comment (vishalinfyom) [inline src/Commands/Publish/PublishUserCommand.php]: format this proper

File: templates/user/user_controller.stub [hunk 2]
```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace $NAMESPACE_CONTROLLER$;
+
+use $NAMESPACE_REQUEST$\CreateUserRequest;
+use $NAMESPACE_REQUEST$\UpdateUserRequest;
+use $NAMESPACE_REPOSITORY$\UserRepository;
+use $NAMESPACE_CONTROLLER$\AppBaseController;
```
Comment (vishalinfyom) [inline templates/user/user_controller.stub]: not used anywhere

File: templates/user/user_repository.stub [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace $NAMESPACE_REPOSITORY$;
+
+use App\User;
+use $NAMESPACE_REPOSITORY$\BaseRepository;
+
+/**
+ * Class UserRepository
+ * @package $NAMESPACE_REPOSITORY$
+ * @version October 8, 2019, 5:51 am UTC
```
Comment (vishalinfyom) [inline templates/user/user_repository.stub]: remove  this

File: src/Commands/Publish/PublishUserCommand.php [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $this->copyView();
+        $this->updateRoutes();
+        $this->updateMenu();
+        $this->publishUserController();
+        $this->publishUserRepository();
```
Comment (ajayinfyom) [inline src/Commands/Publish/PublishUserCommand.php]: there also be check if repository_pattern skip not need to publish UserRepository

Comment (ShaileshInfyom) [inline src/Commands/Publish/PublishUserCommand.php]:
```suggestion
if (config('infyom.laravel_generator.option.repository_pattern')) {
        $this->publishUserRepository();
}
```

File: src/Commands/Publish/PublishUserCommand.php [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+        $this->updateMenu();
+        $this->publishUserController();
+        if (config('infyom.laravel_generator.options.repository_pattern')) {
+            $this->publishUserRepository();
+        }
+        $this->publishCreateUserRequest();
+        $this->publishUpdateUserRequest();
+    }
+
+    private function copyView()
```
Comment (mitulgolakiya) [inline src/Commands/Publish/PublishUserCommand.php]: rename to `copyViews`

File: src/Commands/Publish/PublishUserCommand.php [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+     * @var string
+     */
+    protected $name = 'infyom.publish:user';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Publishes users files';
```
Comment (mitulgolakiya) [inline src/Commands/Publish/PublishUserCommand.php]: `Publishes Users CRUD files`

File: templates/user/create_user_request.stub [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+       $rules = [
+          'name'     => 'required',
+          'email'    => 'required|email|unique:users,email|regex:/^[\w\-\.\+]+\@[a-zA-Z0-9\.\-]+\.[a-zA-z0-9]{2,4}$/',
```
Comment (mitulgolakiya) [inline templates/user/create_user_request.stub]: what is the purpose of this regex

**Pull request merged**